cluster.go

```diff
@@ -10,6 +10,7 @@ import (
 	"github.com/Sirupsen/logrus"
 	"github.com/cenkalti/backoff"
 	"github.com/hailocab/go-hostpool"
+	"golang.org/x/net/context"
```

Comment:
Most of the libraries use `x/net/context`, such as grpc-go, I'm not sure about old version of go, It can be difficult to pass `x/net/context` to `context` argument.